draft-ietf-tls-esni.md

```diff
@@ -430,6 +430,37 @@ the ClientHelloOuter. The server then computes the digest of the reconstructed
 ClientHelloInner. If the digest does not equal OuterExtensions.inner_digest,
 then the server MUST abort the connection with a "decrypt_error" alert.
 
+# The "handshake_padding" Extension {#handshake-padding}
+
+Encrypting the ClientHelloInner may not be sufficient to protect sensitive
+parameters. In particular, the length of the ClientHelloInner, or the length of
+the backend server's response, may leak information about these parameters. The
+client and backend server can use the "handshake_padding" extension in order to
+mitigate this leakage.
+
+A new extension type is defined, which the client MAY include in its
+ClientHello. It MUST NOT offer this extension unless it also advertises support
+for TLS 1.3 or above.
```

Comment:
Meh. I'd drop that second sentence. The extension is perfectly compatible with TLS 1.2. The server will just ignore it as with any other unknown extension.

Reply:
Hmm, the server's reply appears in the EncryptedExtensions message, so the server behavior is ill-defined in TLS<=1.2. I suppose the answer is "who cares"?

Reply:
Removed. I also removed "the server MUST ignore [the] extension if it negotiates TLS 1.2 or below".